.github/workflows/check-vertexai-responses.yml

```diff
@@ -0,0 +1,54 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Check Vertex AI Responses
+
+on: pull_request
+
+jobs:
+  check-version:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Clone mock responses
+      run: scripts/update_vertexai_responses.sh
+    - name: Find cloned and latest versions
+      run: |
+        CLONED=$(git describe --tags)
+        LATEST=$(git tag --sort=v:refname | tail -n1)
+        echo "cloned_tag=$CLONED" >> $GITHUB_ENV
+        echo "latest_tag=$LATEST" >> $GITHUB_ENV
+      working-directory: packages/vertexai/test-utils/vertexai-sdk-test-data
+    - name: Find comment from previous run if exists
+      uses: peter-evans/find-comment@3eae4d37986fb5a8592848f6a574fdf654e61f9e
+      id: fc
+      with:
+        issue-number: ${{github.event.number}}
+        body-includes: Vertex AI Mock Responses Check
+    - name: Comment on PR if newer version is available
+      if: ${{env.cloned_tag != env.latest_tag && !steps.fc.outputs.comment-id}}
+      uses: peter-evans/create-or-update-comment@71345be0265236311c031f5c7866368bd1eff043
+      with:
+        issue-number: ${{github.event.number}}
+        body: >
+          ### Vertex AI Mock Responses Check :warning:
+          
+          A newer major version of the mock responses for Vertex AI unit tests is available.
```

Comment:
Does this only fail with new major versions?

Reply:
The workflow should never "fail". It leaves a comment if the cloned tag is different from the latest available tag, which happens only if there's a newer major version.

Since the script clones the latest tag within a specified major version, the cloned tag will be the latest available tag as long as the specified major version is the latest major version available.

Reply:
Got it. Could you include the new version name in the message to make it easier for folks to update their script?

Reply:
Good idea. I've updated the message and added a dummy tag in the golden files repo to trigger the comment.